### PR TITLE
Add deleteWorkout button

### PR DESCRIPTION
**Pull request by UCSD-Ethan-Ha**
Added delete workout button (x) and implementation.

### CODE REVIEW

File: client/src/tests/WorkoutInput.test.tsx [hunk 1]
```diff
@@ -75,3 +75,41 @@ describe("Create New Workout", () => {
...
+  const newExercise0 = await screen.findByText(/Dumbbell Bench Press/);
+  const newExercise1 = await screen.findByText(/Dumbbell Overhead Press/);
+  const newExercise2 = await screen.findByText(/Lateral Raise/);
+
+  expect(newWorkout).toBeInTheDocument();
+  expect(newExercise0).toBeInTheDocument();
+  expect(newExercise1).toBeInTheDocument();
+  expect(newExercise2).toBeInTheDocument();
+
+  const deleteWorkoutButton = screen.getByText("Delete Workout");
```
Comment (Kobe16) [inline client/src/tests/WorkoutInput.test.tsx]: This test is failing on this line where its trying to find the delete workout button by finding the text "Delete Workout"

However, it looks like the delete workout button that you implemented only has a singular "x" inside the button, and no text called "Delete Workout"

File: client/src/components/WorkoutList.tsx
Comment (Kobe16) [inline client/src/components/WorkoutList.tsx:1]: If you would like to fix the position of the delete button to have it on the same line as the name of the workout, you can use flexbox css styling, like this: 

```
const WorkoutList: React.FC<WorkoutListProps> = ({ workouts, deleteWorkout }) => (
  <div>
    <h2>Workout List</h2>
    {workouts.map((workout, index) => (
      <div key={index} style={{ marginBottom: '20px' }}>
        <div style={{ display: 'flex', flexDirection: 'row' }} >
          <h3>{workout.name}</h3>
          <button onClick={() => deleteWorkout(index)} style={{ margin: "10px", padding: "5px 10px" }}>
            x
          </button>
        </div>
        <ul>
          {workout.exercises.map((exercise, idx) => (
            <li key={idx}>
              {exercise.name} - {exercise.sets} sets, {exercise.reps} reps, {exercise.weight} lbs
            </li>
          ))}
        </ul>
      </div>
    ))}
  </div>
);
```

**Review by Kobe16 — DISMISSED** [review]
Overall, i think the logic and code looks good to me. I just found one minor bug for the delete workout test, and I also added a suggestion on how you could fix the positioning of the delete workout button.

File: client/src/components/WorkoutList.tsx
Comment (UCSD-Ethan-Ha) [reply client/src/components/WorkoutList.tsx:1]: that looks sm better ty

**Review by edolosa — APPROVED** [review]
This implementation looks good on my end, thank you to Kobe for catching that bug and to you Ethan for fixing it promptly. I ran the tests on my side and everything seemed to pass and running a demo I found no issues. Great work here!

**Review by edolosa — APPROVED** [review]

**Pull request merged**